src/finch-mcp-server/README.md

```diff
@@ -0,0 +1,145 @@
+# Finch MCP Server
+
+A Model Context Protocol (MCP) server for Finch that enables generative AI models to build and push container images through finch cli leveraged MCP tools.
+
+## Features
+
+This MCP server acts as a bridge between MCP clients and Finch, allowing generative AI models to build and push container images. The server provides a secure way to interact with Finch, ensuring that the Finch VM is properly initialized and running before performing operations.
+
+## Key Capabilities
+
+- Build container images using Finch
+- Push container images to repositories, including Amazon ECR
+- Check if ECR repositories exist and create them if needed
+- Automatic management of the Finch VM on macos and windows (initialization, starting, etc.)
+- Automatic configuration of ECR credential helpers when needed (only modifies finch.yaml as config.json is automatically handled)
+
+## Prerequisites
+
+1. Install `uv` from [Astral](https://docs.astral.sh/uv/getting-started/installation/) or the [GitHub README](https://github.com/astral-sh/uv#installation)
+2. Install Python using `uv python install 3.10`
+3. Install [Finch](https://github.com/runfinch/finch) on your system
+4. For ECR operations, AWS credentials with permissions to push to ECR repositories and create/describe ECR repositories
+
+## Setup
+
+### Installation
+
+Configure the MCP server in your MCP client configuration (e.g., for Amazon Q Developer CLI, edit `~/.aws/amazonq/mcp.json`):
+
+```json
+{
+  "mcpServers": {
+    "awslabs.finch-mcp-server": {
+      "disabled": false,
+      "timeout": 1000,
+      "command": "uv",
+      "args": [
+        "--directory",
+        "<ABSOLUTE PATH>",
+        "run",
+        "server.py"
+      ],
+      "env": {
+        "AWS_PROFILE": "your-aws-profile",
```

Comment:
may want to consider putting `default` in the example instead of `your-aws-profile`, since `default` should Just Work for most users